templates/zerver/api/incoming-webhooks-walkthrough.md

```diff
@@ -36,6 +36,38 @@ thing, it requires only one fixture,
 }
 ```
 
+If your integration depends on HTTP headers sent by the service you are integrating with,
+then it would be a good idea to store these too. An example of when you would need to do
+this would be if the service sends the event type as a custom http header, like with
+GitHub's webhooks.
+
+To do this,
+
+1. In the folder where your integration's code lives, create a file callled
+`headers.py`.
+
+2. Inside this file, create a python dictionary named `fixtures_to_headers_map`. This
+dictionary should map a fixture's name to a dictionary of headers that you'd like
+to associate the fixture with. Alternatively, the fixture's name could also map to a
+function which should in turn generate a dictionary of headers. Mapping to a function
+would be useful if you need to do some extra processing or contact a third party
+endpoint or etc.
+
+An example of such a dictionary is:
+```python
+fixtures_to_headers_map = {
+    "my_fixture_with_simple_headers_requirements": {"X-Event-Type": "push"},
+    "my_fixture_with_simple_headers_requirements": my_function_to_generate_them,
+}
+
+# the function "my_function_to_generate_them" could be kept in the headers.py file.
+```
+
+Note: If your integration does not depend on HTTP headers, you don't need to follow
+these additional steps. Also, `fixtures_to_headers_map` does not need to map every
+fixture name to a set of headers, just including the fixtures that require them are
+enough.
```

Comment:
This should be at the beginning so that the user can ignore the section before they've read it. :)

Reply:
Right, I definitely agree, I'll make the change. :)